### PR TITLE
[libc] Include empty remove in baremetal stdio.h

### PR DESCRIPTION
**Pull request by petrhosek**
This is required to avoid compilation error in libc++.

See #85335 for more details.

### CODE REVIEW

File: libc/src/stdio/baremetal/remove.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#include "src/stdio/remove.h"
+
+#include "src/__support/common.h"
+
+namespace LIBC_NAMESPACE {
+
+// TODO: See https://github.com/llvm/llvm-project/issues/85335 for more details
+// on why this is needed.
+
+LLVM_LIBC_FUNCTION(int, remove, (const char *)) {
```
Comment (lntue) [inline libc/src/stdio/baremetal/remove.cpp:18]: I feel like we could/should put this in `stdio/generic/`.  WDYT? @nickdesaulniers @michaelrj-google

**Review by michaelrj-google — APPROVED** [review]
LGTM

File: libc/src/stdio/baremetal/remove.cpp (hunk 1 above)
Comment (michaelrj-google) [reply libc/src/stdio/baremetal/remove.cpp:18]: I think we can do that in future if we need a default implementation, but I think it's better to only provide this stub where it's necessary right now.

File: libc/src/stdio/baremetal/remove.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
+//===-- Linux implementation of remove ------------------------------------===//
```
Comment (lntue) [inline libc/src/stdio/baremetal/remove.cpp:1]: nit: Update comment and formatting.

**Review by lntue — APPROVED** [review]

**Comment by nickdesaulniers** [comment]
I'm curious, does this become a problem for other cases of `_LIBCPP_USING_IF_EXISTS` in libcxx/include/cstdio such as fopen, freopen, rename, tmpfile, or tmpnam?  If so, is baremetal going to need stubs for all of the above?

Or is this specifically because `man 3 remove` from stdio.h is conflicting with `std::remove` from `<algorithm>` (and the functions listed above do not)?

**Comment by petrhosek** [comment]
> I'm curious, does this become a problem for other cases of `_LIBCPP_USING_IF_EXISTS` in libcxx/include/cstdio such as fopen, freopen, rename, tmpfile, or tmpnam? If so, is baremetal going to need stubs for all of the above?

I expect this to be only a temporary workaround until we address #85335.

> Or is this specifically because `man 3 remove` from stdio.h is conflicting with `std::remove` from `<algorithm>` (and the functions listed above do not)?

This is only an issue for `remove` because of the declaration with the same name provided by `<algorithm>`.

**Pull request merged**